<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>930</width>
    <height>480</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>930</width>
    <height>480</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>930</width>
    <height>480</height>
   </size>
  </property>
  <property name="cursor">
   <cursorShape>ArrowCursor</cursorShape>
  </property>
  <property name="windowTitle">
   <string>Cache Emulator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupCache">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>911</width>
      <height>291</height>
     </rect>
    </property>
    <property name="title">
     <string>Cache contents</string>
    </property>
    <widget class="QWidget" name="gridLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>891</width>
       <height>261</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout" rowstretch="0,0,0,0,0,0,0,0,0,0,0,0" columnstretch="1,2,3,3,3,3">
      <item row="2" column="0">
       <widget class="QLabel" name="entryId_0">
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>0</string>
        </property>
        <property name="scaledContents">
         <bool>false</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="tag_0">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="data_0_0">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QLabel" name="tag_2">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="entryId_1">
        <property name="text">
         <string>1</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLabel" name="tag_1">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QLabel" name="data_0_1">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="QLabel" name="data_0_2">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="2" column="5">
       <widget class="QLabel" name="data_0_3">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="4" column="2">
       <widget class="QLabel" name="data_1_0">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QLabel" name="data_1_1">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="6" column="2">
       <widget class="QLabel" name="data_2_0">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="entryId_3">
        <property name="text">
         <string>3</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="entryId_4">
        <property name="text">
         <string>4</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QLabel" name="entryId_5">
        <property name="text">
         <string>5</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="entryId_6">
        <property name="text">
         <string>6</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="11" column="0">
       <widget class="QLabel" name="entryId_7">
        <property name="text">
         <string>7</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QLabel" name="tag_3">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QLabel" name="tag_4">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="11" column="2">
       <widget class="QLabel" name="data_7_0">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="11" column="3">
       <widget class="QLabel" name="data_7_1">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="11" column="4">
       <widget class="QLabel" name="data_7_2">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="11" column="5">
       <widget class="QLabel" name="data_7_3">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="10" column="5">
       <widget class="QLabel" name="data_6_3">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="9" column="5">
       <widget class="QLabel" name="data_5_3">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="8" column="5">
       <widget class="QLabel" name="data_4_3">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="7" column="5">
       <widget class="QLabel" name="data_3_3">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="6" column="5">
       <widget class="QLabel" name="data_2_3">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="4" column="5">
       <widget class="QLabel" name="data_1_3">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="4" column="4">
       <widget class="QLabel" name="data_1_2">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="6" column="4">
       <widget class="QLabel" name="data_2_2">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="7" column="4">
       <widget class="QLabel" name="data_3_2">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="8" column="4">
       <widget class="QLabel" name="data_4_2">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="9" column="4">
       <widget class="QLabel" name="data_5_2">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="10" column="4">
       <widget class="QLabel" name="data_6_2">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="6" column="3">
       <widget class="QLabel" name="data_2_1">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="7" column="2">
       <widget class="QLabel" name="data_3_0">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="7" column="3">
       <widget class="QLabel" name="data_3_1">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="8" column="3">
       <widget class="QLabel" name="data_4_1">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="8" column="2">
       <widget class="QLabel" name="data_4_0">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="QLabel" name="tag_5">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="9" column="2">
       <widget class="QLabel" name="data_5_0">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="9" column="3">
       <widget class="QLabel" name="data_5_1">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="10" column="1">
       <widget class="QLabel" name="tag_6">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="10" column="2">
       <widget class="QLabel" name="data_6_0">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="10" column="3">
       <widget class="QLabel" name="data_6_1">
        <property name="text">
         <string>NULL</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="entryId_2">
        <property name="text">
         <string>2</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="11" column="1">
       <widget class="QLabel" name="tag_7">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="entryId_title">
        <property name="font">
         <font>
          <pointsize>8</pointsize>
          <weight>75</weight>
          <bold>true</bold>
          <strikeout>false</strikeout>
         </font>
        </property>
        <property name="text">
         <string>ID</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="tag_title">
        <property name="font">
         <font>
          <pointsize>8</pointsize>
          <weight>75</weight>
          <bold>true</bold>
          <strikeout>false</strikeout>
         </font>
        </property>
        <property name="text">
         <string>Tag</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="label_54">
        <property name="font">
         <font>
          <pointsize>8</pointsize>
          <weight>75</weight>
          <bold>true</bold>
          <strikeout>false</strikeout>
         </font>
        </property>
        <property name="text">
         <string>Data 0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLabel" name="label_55">
        <property name="font">
         <font>
          <pointsize>8</pointsize>
          <weight>75</weight>
          <bold>true</bold>
          <strikeout>false</strikeout>
         </font>
        </property>
        <property name="text">
         <string>Data 1</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QLabel" name="label_56">
        <property name="font">
         <font>
          <pointsize>8</pointsize>
          <weight>75</weight>
          <bold>true</bold>
          <strikeout>false</strikeout>
         </font>
        </property>
        <property name="text">
         <string>Data 2</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QLabel" name="label_57">
        <property name="font">
         <font>
          <pointsize>8</pointsize>
          <weight>75</weight>
          <bold>true</bold>
          <strikeout>false</strikeout>
         </font>
        </property>
        <property name="text">
         <string>Data 3</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupRead">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>310</y>
      <width>301</width>
      <height>51</height>
     </rect>
    </property>
    <property name="title">
     <string>Read</string>
    </property>
    <widget class="QWidget" name="horizontalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>201</width>
       <height>25</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="readLabel">
        <property name="text">
         <string>Address:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit"/>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="readButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>20</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Read</string>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="default">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupWrite">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>370</y>
      <width>301</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Write</string>
    </property>
    <widget class="QWidget" name="horizontalLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>201</width>
       <height>25</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="writeLabel_0">
        <property name="text">
         <string>Address:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_2"/>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="horizontalLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>201</width>
       <height>22</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="writeLabel_1">
        <property name="text">
         <string>Integer: </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_3"/>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="writeButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>20</y>
       <width>75</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>Write</string>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="default">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupInfo">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>310</y>
      <width>221</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>Info</string>
    </property>
    <widget class="QWidget" name="gridLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>201</width>
       <height>111</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_2" columnstretch="0,1">
      <item row="1" column="0">
       <widget class="QLabel" name="infoLabel_1">
        <property name="text">
         <string>Memory capacity:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="memoryCapacity">
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="infoLabel_2">
        <property name="text">
         <string>Words in memory:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="wordsInMemory">
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="infoLabel_0">
        <property name="text">
         <string>Word size:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="word">
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="infoLabel_3">
        <property name="text">
         <string>Cache capacity:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="infoLabel_4">
        <property name="text">
         <string>Words in cache:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLabel" name="cacheCapacity">
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLabel" name="wordsInCache">
        <property name="text">
         <string>Unknown</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupConsole">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>310</y>
      <width>371</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>Console</string>
    </property>
    <widget class="QTextBrowser" name="console">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>351</width>
       <height>111</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>930</width>
     <height>21</height>
    </rect>
   </property>
   <property name="defaultUp">
    <bool>false</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>true</bool>
   </property>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Emulator</string>
    </property>
    <addaction name="actionRandomize_memory"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuAbout"/>
  </widget>
  <action name="actionQuit">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionRandomize_memory">
   <property name="text">
    <string>Randomize memory</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R, F5</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
